db/db_impl/db_impl_compaction_flush.cc

```diff
@@ -2972,6 +3021,30 @@ Status DBImpl::BackgroundFlush(bool* made_progress, JobContext* job_context,
     // This cfd is already referenced
     const FlushRequest& flush_req = PopFirstFromFlushQueue();
     FlushReason flush_reason = flush_req.flush_reason;
+    if (!immutable_db_options_.atomic_flush &&
+        ShouldRescheduleFlushRequestToRetainUDT(flush_req)) {
+      assert(flush_req.cfd_to_max_mem_id_to_persist.size() == 1);
+      ColumnFamilyData* cfd =
+          flush_req.cfd_to_max_mem_id_to_persist.begin()->first;
+      cfd->UnrefAndTryDelete();
```

Comment:
Can this return true, i.e., the `cfd` is deleted?

Reply:
That's a great point! Thank you for catching this, it would otherwise cause segfaults.